### ISSUE TITLE
"Time until stale results are used" also delays non-stale results

### ISSUE THREAD

**Issue by abelbraaksma** [dotnet/fsharp#9200]
At some point, I set the below value to 3s, not really giving it much more thought. Later I thought the VS IDE was unusually sluggish, also on fully loaded solutions, until I remembered this setting and set it back to 200ms. 

My observation is that, after all background tasks are done (icon in bottom-left), the mouse-over for type-info tooltips becomes as slow as that setting. Always. It became responsive when I set it to a low value.

**Repro steps**

1. In this screen, set the value high: 
![image](https://user-images.githubusercontent.com/16015770/82039724-bb0a3500-96a5-11ea-861b-25f1dc39ea24.png)
2. Wait until background processing is finished (it does this when you scroll, swap to a different file, or when you change the above setting, see icon bottom-left, it should stop animating)
3. Hover your mouse over any identifier, it'll *always* take as long as that setting, not only when background processing is still running.

Screencapture 4000ms, after background processing done:

![tooltip-speed4000ms](https://user-images.githubusercontent.com/16015770/82040822-7a132000-96a7-11ea-8be1-9fad07bf7b29.gif)

Now do the opposite, and set the setting to 200ms or so. You'll see that it is now always responsive, regardless of whether stale info is shown.

Screencapture 200ms, after background processing done:

![tooltip-speed200ms](https://user-images.githubusercontent.com/16015770/82040876-8f884a00-96a7-11ea-8bbd-aebc567d7a11.gif)

**Expected behavior**

The setting's name suggests that "stale info" should kick in whenever background processing isn't finished. But if it is, I'd expect it to be immediate (i.e., 100-200ms).

In other words, I expect that: if, and only if, background processing isn't finished, show stale data. Otherwise, show current data and show it immediately.

**Actual behavior**

It turns out to be *always* slow, regardless whether it shows actual, or cached/stale info. Also, I couldn't get it to show stale info (change a type, it occasionally shows `'a -> obj` and then later changes to `'a -> 'b`, perhaps that is what "stale" means here?). Regardless, it simply always delays the tooltip, which I think it shouldn't, or the name of the setting should be changed.

It affects *all tooltips*, that is, errors, type info etc.

**Known workarounds**

Set it to a low value and accept that stale info may be shown, even in a quick tooltip.

* VS IDE up-to-date, non-preview version.

**Comment by abelbraaksma** [comment]
This may be a good moment to share that compiling my project can take between 30s and 2 minutes, depending on how much has changed. But waiting for tooltips can easily take up to 5 minutes or more. While I was making the following screen capture, and while I am still typing this, background processing is still going. The stale setting from above is 200ms.

If you watch this, load the page, go for a coffee, drink it, and then come back, you'll see the tooltip appear (first one around 3300 frames, 2nd one at 4300 frames (each frame is 50ms - 100ms). You get the idea ;).

![tooltip-no-stale-data](https://user-images.githubusercontent.com/16015770/82042239-da0ac600-96a9-11ea-8b3d-2f052dd3b86a.gif)

Stale data could help here, but that doesn't appear to work. Note that this project was opened the day before. It was compiled (but with three errors). No edits were done before any of these screen grabs. The file in the above screen grab is "at the end" of the project.

**Comment by dsyme** [comment]
I'm going to close this issue.  People very rarely fiddle with this setting, and @TIHan did a huge amount of work on performance and responsiveness (getting rid of the reactor queue)

**Comment by abelbraaksma** [comment]
I think this actually improved a lot. I haven't seen this behavior in a while. Now that VS2022 is 64 bits and with all the improvements in the language service, it's very well possible that the cause simply vanished.

Thanks for taking the time to go over these old issues!
